### DISCUSSION TITLE
PyInstaller on ARM based macbook creates program that just segfaults.

### DISCUSSION THREAD

**Discussion by Magicrafter13** [pyinstaller/pyinstaller#7814] [Help] [answered]
I'm running the program through a virtual environment, in which I only have the following packages installed: `PyQt6`, `sftpretty`, and `pyinstaller`.

The program runs perfectly fine just with `venv/bin/python3 main.py`. It also runs perfectly fine on Windows and Linux (x86), and pyinstaller successfully creates standalone executables for those platforms. But when I try to do this on my ARM-based macbook, though pyinstaller runs without any noticeable (to me) errors, the app it produces simply opens up with a console window showing that it had a segmentation fault, and gives no other useful output.

This is the output from a clean build with pyinstaller:
https://hive.matthewrease.net/s/6wDAQwmxnkffKC8 (this is a Nextcloud shared-file link)

On earlier attempts, where I was accidentally using the system-wide installed packages and pyinstaller, and not the ones in the venv, it gave a different error when running the executable:
```
Fatal Python error: init_fs_encoding: failed to get the Python codec of the filesystem encoding
```
Looking into it online, I saw mention of outdated packages, and this macbook did originally come with an older version of Python3 before manually being updated to 3.11, hence I made sure it was using the venv, and when it did this error was no longer present. As such I'm not sure if it is important to the true issue at hand, but figured I should add it just in case.

**Comment by bwoodsend** [comment, marked as answer]
You're probably being hit by https://github.com/pyinstaller/pyinstaller/issues/7789?
